streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -520,11 +522,24 @@ public Long apply(K aggKey, V value, Long aggregate) {
     }
 
     @Override
-    public     KTable<K, Long> countByKey(Serializer<K> keySerializer,
-                                          Serializer<Long> aggValueSerializer,
-                                          Deserializer<K> keyDeserializer,
-                                          Deserializer<Long> aggValueDeserializer,
-                                          String name) {
+    public KTable<K, Long> countByKey(Serializer<K> keySerializer,
+                                      Deserializer<K> keyDeserializer,
+                                      String name) {
+        return countByKey(
+                keySerializer,
+                new LongSerializer(),
```

Comment:
For `KStreamImpl`: Similar to what @ijuma said at https://github.com/apache/kafka/pull/1066, why are we creating new `LongSerializer` and `LongDeserializer` instances for every call to `count()`?  Can't we reuse the same (e.g. cached) instances?